notebooks/video_clips/yolo4.py

```diff
@@ -0,0 +1,201 @@
+import sys
+import cv2
+import argparse
+import random
+import time
+import os
+import os.path as osp
+
+import urllib3
+from pathlib import Path
+import hashlib
+from tqdm import tqdm
+
+# source: https://github.com/kingardor/YOLOv4-OpenCV-CUDA-DNN
+class YOLOv4:
+
+    def __init__(self,args=None):
+        """ Method called when object of this class is created. """
+
+        self.args = None
+        self.net = None
+        self.names = None
+
+        self.args = self.parse_arguments() if args is None else args
+        self.initialize_network()
+        self.run_inference()
+
+    def parse_arguments(self):
+        """ Method to parse arguments using argparser. """
+
+        parser = argparse.ArgumentParser(description='Object Detection using YOLOv4 and OpenCV4')
+        parser.add_argument('--image', type=str, default='', help='Path to use images')
+        parser.add_argument('--stream', type=str, default='', help='Path to use video stream')
+        parser.add_argument('--cfg', type=str, default='models/yolov4.cfg', help='Path to cfg to use')
+        parser.add_argument('--weights', type=str, default='models/yolov4.weights', help='Path to weights to use')
+        parser.add_argument('--namesfile', type=str, default='models/coco.names', help='Path to names to use')
+        parser.add_argument('--input_size', type=int, default=416, help='Input size')
+        parser.add_argument('--use_gpu', default=False, action='store_true', help='To use NVIDIA GPU or not')
+        parser.add_argument('--outdir', type=str, default="video", help='Location to put the output')
+        parser.add_argument('--no-squash-detections', action='store_true', help="Fail if detections ouput exists") 
+
+        return parser.parse_args()
+
+    def initialize_network(self):
+        """ Method to initialize and load the model. """
+
+        self.net = cv2.dnn_DetectionModel(self.args.cfg, self.args.weights)

```

Comment:
We should provide more bits of information on the core of this wrapper.

Do we want to clarify to the reader that till v4 the official repo was not from uralytics, and here we are creating a cv2 model. https://github.com/AlexeyAB/darknet .

Also this link would be useful to curious folks. https://arxiv.org/abs/2004.10934

A segway into cv2 dnn would also be like a next step to explore more. https://docs.opencv.org/4.x/da/d9d/tutorial_dnn_yolo.html